### PR TITLE
New blog article: kamailio-consul-service-discovery

### PR DESCRIPTION
**Pull request by tranchitella**

### CODE REVIEW

File: content/blog/kamailio-consul-service-discovery.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+Dynamic scaling brought additional complexity in the configuration of our Kamailio components. Having a cloud-native solution meant that we had to find a way to make SBCs and routers auto-configure themselves without human intervention or static configurations, which will bound the size of our architecture.
+
+We wanted to deploy Wazo on different kinds of platforms, not only containerized with Kubernetes but also with Docker compose or on Virtual machines using Ansible recipes. We had to choose a tool that will help us in service discovery and the flexible configuration of our components. That's where we decided that [HashiCorp Consul](www.consul.io) was the right tool to use.
+
+## What is Consul?
+
+The [official](www.consul.io) definition of Consul is:
+
+```Consul is a service networking solution to connect and secure services across any runtime platform and public or private cloud```
```
Comment (rgarrigue) [inline content/blog/kamailio-consul-service-discovery.md]: nitpicking, > citation instead of ```codeblock``` ?

**Comment by fredericlepied** [comment]
@tranchitella feel free to revert my changes if misunderstood something...

**Comment by sboily** [comment]
This is nice article, and in time about what we did the last week with @safchain and @lebauce around scalability with Asterisk, consul and AMQP with a kamailio.

**Comment by safchain** [comment]
Nice article thanks @tranchitella

File: content/blog/kamailio-consul-service-discovery.md (hunk 1 above)
Comment (alexsosic) [reply content/blog/kamailio-consul-service-discovery.md]: @rgarrigue `docker_entrypoint.sh` is the "entrypoint" which launches the executables. Then we just do a `trap exit_script SIGINT SIGTERM` ;)
Take a look at that bash script and you should understand how we handle the unregister.

Comment (tranchitella) [reply content/blog/kamailio-consul-service-discovery.md]: @rgarrigue  I would add that we use auto-deregister if the health check fails above a threshold.

**Pull request merged**